### PR TITLE
Cache identify results, use cache when source is kbfs

### PR DESCRIPTION
**Pull request by patrickxb**
r? @maxtaco 

First part of what we discussed, caches all identify results for 5 minutes.  If source is kbfs, then it checks the cache on identify call.

Tested with local kbfs, identify calls correctly short-circuited.

Note:  this does not contain the self-identify changes we discussed.

### CODE REVIEW

File: go/service/identify.go [hunk 1]
```diff
@@ -24,30 +28,48 @@ type RemoteIdentifyUI struct {
...
 	}
 }
 
 func (h *IdentifyHandler) Identify(_ context.Context, arg keybase1.IdentifyArg) (keybase1.IdentifyRes, error) {
-	res, err := h.identify(arg.SessionID, arg, true)
-	if err != nil {
-		return keybase1.IdentifyRes{}, err
+	if arg.Source == keybase1.IdentifySource_KBFS {
+		h.G().Log.Debug("KBFS Identify: checking result cache for %q", arg.UserAssertion)
+		x, err := h.resultCache.Get(arg.UserAssertion)
```
Comment (maxtaco) [inline go/service/identify.go]: is this goroutine safe?

Comment (patrickxb) [inline go/service/identify.go]: yes

**Comment by maxtaco** [comment]
i think is fine temporarily, but we probably want to push the code down into the identify engine longer term.

**Comment by maxtaco** [comment]
:+1: for now, let's reconsider a better solution when we have a moment.

**Comment by patrickxb** [comment]
Understood.  Trying to do this in pieces so problems are alleviated ASAP.

**Pull request merged**